tests/tests_app/core/test_lightning_work.py

```diff
@@ -280,3 +280,37 @@ def run(self, *args, **kwargs):
     w.run(1, [2], (3, 4), {"1": "3"})
     assert len(w._calls) == 2
     assert w._calls["0d824f7"] == {"ret": None}
+
+
+def test_work_cloud_build_config_provided():
+
+    assert isinstance(LightningWork.cloud_build_config, property)
+    assert LightningWork.cloud_build_config.fset is not None
+
+    class Work(LightningWork):
+        def __init__(self):
+            super().__init__()
+            self.cloud_build_config = BuildConfig(image="ghcr.io/gridai/base-images:v1.8-cpu")
+
+        def run(self, *args, **kwargs):
+            pass
+
+    w = Work()
+    w.run()
+
+
+def test_work_local_build_config_provided():
```

Comment:
the tests looks same to be.. What's the difference?

Reply:
The difference is local vs. cloud build config. To this point I would suggest not making this test about build config, but about ANY setter getter. This is an alternative test more general:


```py
def test_work_setattr_for_property_getters_and_setters():
    """Test that Work's custom `__setattr__` works with property getters and setters."""

    class Work(LightningWork):
        def __init__(self):
            super().__init__()
            self._attribute = ""

        @property
        def my_property(self):
            return self._attribute = ""

        @my_property.setter
        def my_property(self, value):
            self._attribute = value

        def run(self, *args, **kwargs):
            self.my_property = "new_value"
            assert self._attribute == "new_value"

            # try a non-json serializable object
            self.my_property = BuildConfig(image="ghcr.io/gridai/base-images:v1.8-cpu")
            assert isinstance(self._attribute, BuildConfig)


    w = Work()
    w.run()
```
(Sketch, might have a typo)


I'm also fine with the current tests.

Reply:
I like the idea of a general setter test as well but i had originally decided against it because it seemed to be out-of-scope of the issue that this PR intends to solve. Also, given that all the other custom setters are implemented in the same manner, if these two tests pass, we can safely assume all other custom setters are being used as expected.

Reply:
Should this be resolved?